botocore/client.py

```diff
@@ -174,8 +171,46 @@ def waiter_names(self):
     def _load_service_model(self, service_name):
         json_model = self._loader.load_service_model('aws/%s' % service_name)
         service_model = ServiceModel(json_model, service_name=service_name)
+        self._register_retries(service_model)
         return service_model
 
+    def _register_retries(self, service_model):
```

Comment:
can we break this up into multiple methods.  I've been trying to keep the client method pretty clean and avoid long methods where possible.  I'd suggest instead of a comment block, extract a method instead.